pysmt/rewritings.py

```diff
@@ -832,3 +832,29 @@ def disjunctive_partition(formula):
                 to_process += cur.args()
             else:
                 yield cur
+
+
+def propagate_toplevel(formula, env=None, do_simplify=True):
+    """ Propagates the toplevel definitions and returns an equivalent formula.
+    It considers two kinds of definitions:
+    1) variable = constant
+    2) variable = variable
+    """
+    if env is None:
+        import pysmt.environment
+        env = pysmt.environment.get_env()
+    mgr = env.formula_manager
+    sigma = {}
+    for c in conjunctive_partition(formula):
+        if c.is_equals():
+            l, r = sorted(c.args(), key=hash)
```

Comment:
Two elements could have the same hash and so you might loose the ordering you want (This is not true with pysmt formulae, but it creates confusion). You can use the `FNode.node_id` function instead of the `hash` function. The id of two pysmt formulae are guaranteed to be different if the two objects are different.

Reply:
Ah, I didn't know that. 
Thanks for pointing this out.